clippy_lints/src/methods/mod.rs

```diff
@@ -1709,6 +1753,28 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
     }
 }
 
+/// Checks for the `ENTRY_OR_INSERT_WITH_DEFAULT` lint.
+fn lint_entry_or_insert_with_default(cx: &LateContext<'tcx>, span: Span, args: &'tcx [hir::Expr<'_>]) {
+    if_chain! {
+        let ty = cx.typeck_results().expr_ty(&args[0]);
+        if match_type(cx, ty, &paths::BTREEMAP_ENTRY) || match_type(cx, ty, &paths::HASHMAP_ENTRY);
+        if let hir::ExprKind::Path(arg_path) = &args[1].kind;
+        if let hir::def::Res::Def(_, arg_def_id) = cx.qpath_res(&arg_path, args[1].hir_id);
+        if match_def_path(cx, arg_def_id, &paths::DEFAULT_TRAIT_METHOD);
+        then {
+            span_lint_and_sugg(
+                cx,
+                ENTRY_OR_INSERT_WITH_DEFAULT,
+                span,
+                "use of `or_insert_with` with `Default::default`",
+                "replace with",
+                "or_default()".to_owned(),
+                Applicability::MachineApplicable,
+            );
+        }
+    }
+}
+
```

Comment:
I wonder if this function should be merged with `check_unwrap_or_default` a few lines below

Reply:
Not sure, but I would need some help figuring out how, to be honest.

Reply:
Yes, this should be merged. You'll want to hoist that function out and make the `unwrap_or` part a parameter